src/Microsoft.DotNet.Darc/src/DarcLib/Models/MsBuildPropsFile.cs

```diff
@@ -48,6 +49,12 @@ protected MsBuildPropsFile(bool? orderPropertiesAscending)
 
     public void SerializeToXml(string path)
     {
+        var parentDir = Path.GetDirectoryName(path) ?? throw new ArgumentException($"'{path}' is not a valid path.");
```

Comment:
Nit: the `parentDir` is a bit misleading name as it's actually the name of the directory, not it's parent.

Reply:
Renamed